src/main/java/uk/ac/ox/oxfish/biology/growers/FadAwareLogisticGrower.java

```diff
@@ -0,0 +1,211 @@
+/*
+ *  POSEIDON, an agent-based model of fisheries
+ *  Copyright (C) 2020  CoHESyS Lab cohesys.lab@gmail.com
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package uk.ac.ox.oxfish.biology.growers;
+
+import com.google.common.collect.ImmutableList;
+import ec.util.MersenneTwisterFast;
+import org.jetbrains.annotations.NotNull;
+import sim.engine.SimState;
+import sim.engine.Steppable;
+import sim.engine.Stoppable;
+import uk.ac.ox.oxfish.biology.BiomassLocalBiology;
+import uk.ac.ox.oxfish.biology.Species;
+import uk.ac.ox.oxfish.fisher.equipment.fads.Fad;
+import uk.ac.ox.oxfish.geography.fads.FadMap;
+import uk.ac.ox.oxfish.model.FishState;
+import uk.ac.ox.oxfish.model.Startable;
+import uk.ac.ox.oxfish.model.StepOrder;
+import uk.ac.ox.oxfish.model.data.monitors.Monitor;
+import uk.ac.ox.oxfish.model.data.monitors.accumulators.Accumulator;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.Streams.concat;
+import static com.google.common.collect.Streams.stream;
+import static uk.ac.ox.oxfish.biology.growers.CommonLogisticGrower.allocateBiomassProportionally;
+import static uk.ac.ox.oxfish.biology.growers.DerisoSchnuteCommonGrower.allocateBiomassAtRandom;
+import static uk.ac.ox.oxfish.biology.growers.IndependentLogisticBiomassGrower.logisticRecruitment;
+import static uk.ac.ox.oxfish.fisher.equipment.fads.FadManagerUtils.maybeGetFadManager;
+import static uk.ac.ox.oxfish.model.StepOrder.BIOLOGY_PHASE;
+import static uk.ac.ox.oxfish.model.StepOrder.DATA_RESET;
+import static uk.ac.ox.oxfish.utility.FishStateUtilities.EPSILON;
+
+/**
+ * The SchaeferLogisticGrower is like a CommonLogisticGrower, but calculates growth by using the
+ * memorized biomass from the previous year instead of using the current biomass.
+ * This grower takes FAD biomass into account as part of the total memorized biomass, but only
+ * redistributes new biomass in ocean cells. When the growth function is called, it also adds back
+ * biomass that was lost by FADs drifting out of habitable ocean cells. If no FADs are present, it
+ * works just like a CommonLogisticGrower (except of course for the memorized biomass bit).
```

Comment:
so, to be clear, we need the repetition because:
1) We need a a biomassLost counter
2) We need to add distributional weights
correct?

Reply:
It needs to:

- Count the biomass under FADs as part of the current biomass used for growth, but not redistribute new biomass under FADs.
- Add back the biomass lost by FADs beaching or travelling outside the habitable zone.
- Be able to work in a Shaefer-like way (i.e., use beginning-of-year for growth.

I think we don't *need* the repetition and that all of this could be wrapped up in a single `CommonLogisticGrower` class; but it needs more testing before we fuse them...

Reply:
cool!